pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -114,6 +123,11 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 	if len(req.Namespace) == 0 {
 		req.Namespace = svcacct.Namespace
 	}
+	if len(req.UID) == 0 {
+		req.UID = svcacct.UID
+	} else if req.UID != svcacct.UID {
+		warning.AddWarning(ctx, "", fmt.Sprintf("the UID in the token request (%s) does not match the UID of the service account (%s) but TokenRequestServiceAccountUIDValidation is not enabled. In future, this will return a conflict error", req.UID, svcacct.UID))
```

Comment:
```suggestion
		warning.AddWarning(ctx, "", fmt.Sprintf("the UID in the token request (%s) does not match the UID of the service account (%s) but TokenRequestServiceAccountUIDValidation is not enabled. In the future, this will return a conflict error", req.UID, svcacct.UID))
```